src/Analyzer/Passes/QueryAnalysisPass.cpp

```diff
@@ -1396,6 +1396,8 @@ class QueryAnalyzer
     /// Lambdas that are currently in resolve process
     std::unordered_set<IQueryTreeNode *> lambdas_in_resolve_process;
 
+    std::unordered_set<String> cte_in_resolve_process;
```

Comment:
Technically can be std::string_view, because query node cannot be destroyed during `resolveQuery` or `resolveUnion`.